lib/Authen/SASL/Perl/XOAUTH2.pm

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2025 Aditya Garg <gargaditya08@live.com>
+# Copyright (c) 2025 Julian Swagemakers <julian@swagemakers.org>
+# All rights reserved.
+# This program is free software; you can redistribute it and/or
+# modify it under the same terms as Perl itself.
+
+package Authen::SASL::Perl::XOAUTH2;
+
+use strict;
+use vars qw(@ISA);
+
+@ISA     = qw(Authen::SASL::Perl);
+
+my %secflags = (
+    noanonymous => 1,
+);
+
+sub _order { 1 }
+
+sub _secflags {
+    shift;
+    scalar grep { $secflags{$_} } @_;
+}
+
+sub mechanism { 'XOAUTH2' }
+
+
+sub client_start {
+    my $self = shift;
+    $self->{stage} = 0;
+
+    # "user=" {User} "^Aauth=Bearer " {Access Token} "^A^A"
+    # https://developers.google.com/gmail/imap/xoauth2-protocol#initial_client_response
+    my $username = $self->_call('user');
+    my $token    = $self->_call('pass'); # OAuth 2.0 access token
+    my $auth_string = "user=$username\001auth=Bearer $token\001\001";
```

Comment:
In the previous implementation, there was base64 encoding of the auth string. In this iteration that's no longer the case. Is that correct?